### PR TITLE
LPS-201012 | Poshi Automation | Master

### PR DESCRIPTION
**Pull request by rafaellajordao**
# LPS-197163 - Product QA | Test Automation Creation

Hi! :)
This pr is corresponding to this ticket: [LPS-197163](https://liferay.atlassian.net/browse/LPS-197163) → [LPS-201012](https://liferay.atlassian.net/browse/LPS-201012)

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/WebContentAccessibility.testcase [hunk 1]
```diff
@@ -227,4 +227,70 @@ definition {
...
+
+		task ("When searching for a term on the web content admin page") {
+			WebContentNavigator.openWebContentAdmin(siteURLKey = ${randomSiteName});
+
+			Search.searchCP(searchTerm = "Years");
+
+			KeyPress(
+				key_breadcrumbName = "Home",
+				locator1 = "Breadcrumb#BREADCRUMB_ENTRY_LINK",
+				value1 = "\TAB");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/WebContentAccessibility.testcase:258]: This element should be previous focused one not Home

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/WebContentAccessibility.testcase [hunk 2]
```diff
@@ -227,4 +227,70 @@ definition {
...
+
+			KeyPress(
+				key_breadcrumbName = "Home",
+				locator1 = "Breadcrumb#BREADCRUMB_ENTRY_LINK",
+				value1 = "\ENTER");
+
+			AssertTextEquals.assertPartialText(
+				key_webContentTitle = "WC Folder Name",
+				locator1 = "WC#ENTRY_LIST_TITLE",
+				value1 = "WC Folder Name");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/WebContentAccessibility.testcase:268]:
```
Breadcrumb.viewActiveEntry

WebContentAdmin.viewEntryTitle
```

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/WebContentAccessibility.testcase [hunk 3]
```diff
@@ -227,4 +227,70 @@ definition {
...
+
+		task ("Then it is possible to access the file path using the keyboard") {
+			WebContentNavigator.openWebContentAdmin(siteURLKey = ${randomSiteName});
+
+			Search.searchCP(searchTerm = "Years");
+
+			KeyPress(
+				key_breadcrumbName = "WC Folder Name",
+				locator1 = "Breadcrumb#BREADCRUMB_ENTRY_LINK",
+				value1 = "\TAB");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/WebContentAccessibility.testcase:279]: Same as above

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/WebContentAccessibility.testcase [hunk 4]
```diff
@@ -227,4 +227,70 @@ definition {
...
+				locator1 = "Breadcrumb#BREADCRUMB_ENTRY_LINK",
+				value1 = "\ENTER");
+
+			AssertElementPresent(
+				key_breadcrumbName = "Home",
+				locator1 = "Breadcrumb#BREADCRUMB_ENTRY");
+
+			AssertElementPresent(
+				locator1 = "Breadcrumb#BREADCRUMB_ENTRY_ACTIVE_ENTRY",
+				value1 = "WC Folder Name");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/WebContentAccessibility.testcase:292]: Breadcrumb.viewActiveEntry

WebContentAdmin.viewEntryTitle

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContent.testcase [hunk 5]
```diff
@@ -1767,6 +1767,60 @@ definition {
...
+		}
+
+		task ("When searching for a term on the web content admin page") {
+			WebContentNavigator.openWebContentAdmin(siteURLKey = "test-site-name");
+
+			Search.searchCP(searchTerm = "Years");
+
+			Click(
+				key_breadcrumbName = "Home",
+				locator1 = "Breadcrumb#BREADCRUMB_ENTRY_LINK");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContent.testcase:1797]: Add a macro named Breadcrumb.gotoEntry

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContent.testcase [hunk 6]
```diff
@@ -1767,6 +1767,60 @@ definition {
...
+
+			AssertTextEquals.assertPartialText(
+				key_webContentTitle = "WC Folder Name",
+				locator1 = "WC#ENTRY_LIST_TITLE",
+				value1 = "WC Folder Name");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContent.testcase:1802]: Same as above

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContent.testcase [hunk 7]
```diff
@@ -1767,6 +1767,60 @@ definition {
...
+				key_breadcrumbName = "WC Folder Name",
+				locator1 = "Breadcrumb#BREADCRUMB_ENTRY_LINK");
+
+			AssertElementPresent(
+				key_breadcrumbName = "Home",
+				locator1 = "Breadcrumb#BREADCRUMB_ENTRY");
+
+			AssertElementPresent(
+				locator1 = "Breadcrumb#BREADCRUMB_ENTRY_ACTIVE_ENTRY",
+				value1 = "WC Folder Name");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContent.testcase:1820]: Same as above

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/WebContentAccessibility.testcase [hunk 8]
```diff
@@ -227,4 +227,70 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-197163. When displaying the search results, the file path can be accessed via keyboard."
+	@priority = 4
+	test ViewFilePathInteractiveElementViaKeyboard {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/WebContentAccessibility.testcase:232]: Move to WebContentAdminWithKeyboard.testcase then rename to NavigateByBreadcrumbInSearchResultsViaKeyboard

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContent.testcase [hunk 9]
```diff
@@ -1767,6 +1767,60 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-197163. When displaying the search results, the file path must be an interactive element."
+	@priority = 4
+	test ViewFilePathInteractiveElement {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContent.testcase:1772]: Move to WebContentUI.testcase then rename to NavigateByBreadcrumbInSearchResults

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/WebContentAccessibility.testcase [hunk 10]
```diff
@@ -227,4 +227,70 @@ definition {
...
+				folderName = "WC Folder Name",
+				groupName = ${randomSiteName},
+				title = "Web Content Title");
+		}
+
+		task ("When searching for a term on the web content admin page") {
+			WebContentNavigator.openWebContentAdmin(siteURLKey = ${randomSiteName});
+
+			Search.searchCP(searchTerm = "Years");
+
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/WebContentAccessibility.testcase:254]: Then block

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/WebContentAccessibility.testcase [hunk 11]
```diff
@@ -227,4 +227,70 @@ definition {
...
+				locator1 = "Breadcrumb#BREADCRUMB_ENTRY_LINK",
+				value1 = "\ENTER");
+
+			AssertTextEquals.assertPartialText(
+				key_webContentTitle = "WC Folder Name",
+				locator1 = "WC#ENTRY_LIST_TITLE",
+				value1 = "WC Folder Name");
+		}
+
+		task ("Then it is possible to access the file path using the keyboard") {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/WebContentAccessibility.testcase:271]: When block

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/WebContentAccessibility.testcase [hunk 12]
```diff
@@ -227,4 +227,70 @@ definition {
...
+			WebContentNavigator.openWebContentAdmin(siteURLKey = ${randomSiteName});
+
+			Search.searchCP(searchTerm = "Years");
+
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/WebContentAccessibility.testcase:275]: Then block

**Review by Tim-Cao — CHANGES_REQUESTED** [review]
Hi @rafaellajordao, would you mind modifying according to comments? Thanks

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContent.testcase (hunk 9 above)
Comment (rafaellajordao) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContent.testcase:1772]: done!

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContent.testcase (hunk 5 above)
Comment (rafaellajordao) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContent.testcase:1797]: done!

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/WebContentAccessibility.testcase (hunk 8 above)
Comment (rafaellajordao) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/WebContentAccessibility.testcase:232]: done!

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/WebContentAccessibility.testcase (hunk 2 above)
Comment (rafaellajordao) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/WebContentAccessibility.testcase:268]: done

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/WebContentAccessibility.testcase (hunk 1 above)
Comment (rafaellajordao) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/WebContentAccessibility.testcase:258]: I don't understand, could you explain to me what I should do?

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/WebContentAccessibility.testcase (hunk 3 above)
Comment (rafaellajordao) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/accessiblity/WebContentAccessibility.testcase:279]: I don't understand, could you explain to me what I should do?

**Comment by Tim-Cao** [comment]
@rafaellajordao, see changes https://github.com/liferay-echo/liferay-portal/pull/14330

**Comment by Tim-Cao** [comment]
ci:close

**Pull request closed without merging**